frontend/components/NavDrawerCategoriesTree/components/CategoriesItem/index.jsx

```diff
@@ -86,13 +118,25 @@ const CategoriesItem = ({
 
   const handleClick = useCallback(() => {
     setIsOpen(!isOpen);
+    // ###BOA### 
+    // Logic for page switcher (if used with @shopgate-project/page-switcher)
+    if (level === 0) {
+      setisPageSwitcherInitalState(false);
+    }
+    // // ###EOA### 
   }, [isOpen]);
 
   const handleOpenCategory = useCallback(() => {
     openCategory(categoryId);
   }, [categoryId, openCategory]);
 
   const buttonRight = useMemo(() => {
+    // ###BOA### 
+    // Logic for page switcher (if used with @shopgate-project/page-switcher)
+    if (pageSwitcher && isPageSwitcherInitalState && level === 0) {
+      setIsOpen(true);
```

Comment:
i guess this will be removed eventually (see comment above), but this is a bad implementation since it causes unnecessary re-renders.

it will be executed after `pageSwitcher` turns `true` because we received the required data to determine the final value. this state change causes a re-render of the component which is not really needed.